File: src/main/java/social/alone/server/config/AppProperties.java
package social.alone.server.config;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@ConfigurationProperties(prefix = "app")
@Component
@Getter
public class AppProperties {

  private final Auth auth = new Auth();
  private final OAuth2 oauth2 = new OAuth2();
  private final Link link = new Link();
  private final Slack slack = new Slack();
  private final Sentry sentry = new Sentry();

  // TODO lombok fix
  public static class Auth {
    private String tokenSecret;
    private long tokenExpirationMsec;

    public String getTokenSecret() {
      return tokenSecret;
    }

    public void setTokenSecret(String tokenSecret) {
      this.tokenSecret = tokenSecret;
    }

    public long getTokenExpirationMsec() {
      return tokenExpirationMsec;
    }

    public void setTokenExpirationMsec(long tokenExpirationMsec) {
      this.tokenExpirationMsec = tokenExpirationMsec;
    }
  }

  // TODO lombok fix
  @Data
  public static final class OAuth2 {

    private String clientId;
    private String clientSecret;

    private List<String> authorizedRedirectUris = new ArrayList<>();

    public List<String> getAuthorizedRedirectUris() {
      return authorizedRedirectUris;
    }

  }

  @Data
  public static final class Link {
    private String host;

  }

  @Data
  public static final class Slack {
    private String channel;
  }


  @Data
  public static final class Sentry {
    private String dsn;
  }


  public Auth getAuth() {
    return auth;
  }

  public OAuth2 getOauth2() {
    return oauth2;
  }
}

File: src/main/kotlin/social/alone/server/config/AppProperties.kt
package social.alone.server.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.util.*

@ConfigurationProperties(prefix = "app")
@Component
data class AppProperties(
        val auth:Auth = Auth(),
        val oauth2:OAuth2 = OAuth2(),
        val link:Link = Link(),
        val slack:Slack = Slack(),
        val sentry:Sentry = Sentry()
) {

    data class Auth (var tokenSecret: String? = null, var tokenExpirationMsec: Long = 3600000){
    }


    data class OAuth2 (
        val clientId: String? = null,
        val clientSecret: String? = null,
        val authorizedRedirectUris: List<String> = ArrayList()
    )

    data class Link (var host: String? = null)

    data class Slack(var channel: String? = null)

    data class Sentry(var dsn: String? = null)
}
